js/router.js

```diff
@@ -48,9 +48,22 @@ module.exports = Backbone.Router.extend({
         this.navigate(translatedRoute, { trigger: true });
       });
     });
-    history.size = -1;
-    history.position = -1;
-    history.action = 'default';
+    
+    var originalHistoryBack = history.back;
+    history.back = function() {
```

Comment:
It's slightly dangerous to overwrite such core functions. At the very least be sure to pass the arguments through and also maintain the return value. As of now, the core back and forward functions don't take any args, nor do they return anything, but we should make this as future-proof as possible:

```javascript
var originalHistoryForward = history.forward;
history.forward = function() {
    historyAction = 'forward';
    return originalHistoryForward(arguments);
 }
```